packages/core/src/service/helpers/utils/tree-relations-qb-joiner.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Verifies if a relation has already been joined in a query builder to prevent duplicate joins.
+ * This method ensures query efficiency and correctness by maintaining unique joins within the query builder.
+ *
+ * @param {SelectQueryBuilder<T>} qb The query builder instance where the joins are being added.
+ * @param {string} alias The join alias to check for uniqueness. This alias is used to determine if the relation
+ *                       has already been joined to avoid adding duplicate join statements.
+ * @returns boolean Returns true if the relation has already been joined (based on the alias), false otherwise.
+ * @template T extends VendureEntity The entity type for which the query builder is configured.
+ */
```

Comment:
Looks like this doc block is out of place?

Reply:
Yeah, I don't know how to do it here. If you have recommendations, I'd be glad to change it and maybe I need to reduce the amount of text...

Reply:
> Looks like this doc block is out of place?

I deleted the documentation that was accidentally copied and updated the documentation I was adding for the method. Check it out, please.